src/tests/unit/tests/electron/platform/android/android-setup-telemetry-sender.test.ts

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { TelemetryEventHandler } from 'background/telemetry/telemetry-event-handler';
+import { TelemetryEventSource, TriggeredByNotApplicable } from 'common/extension-telemetry-events';
+import {
+    AndroidSetupStepTelemetryData,
+    DEVICE_SETUP_STEP,
+} from 'electron/common/electron-telemetry-events';
+import { AndroidSetupStore } from 'electron/flux/store/android-setup-store';
+import { AndroidSetupTelemetrySender } from 'electron/platform/android/android-setup-telemetry-sender';
+import { AndroidSetupStepId } from 'electron/platform/android/setup/android-setup-step-id';
+import { It, Mock, MockBehavior, Times } from 'typemoq';
+
+describe('AndroidSetupTelemetrySender', () => {
+    it('sends appropriate telemetry events when steps change', () => {
+        testTelemetryEvents(
+            ['detect-adb', 'detect-devices', 'detect-permissions'],
+            [20, 400, 415],
+            [
+                {
+                    prevStep: null,
+                    newStep: 'detect-adb',
+                    prevDuration: 0,
+                },
+                {
+                    prevStep: 'detect-adb',
+                    newStep: 'detect-devices',
+                    prevDuration: 380,
+                },
+                {
+                    prevStep: 'detect-devices',
+                    newStep: 'detect-permissions',
+                    prevDuration: 15,
+                },
+            ],
+        );
+    });
+
+    it('does not send event if step has not changed', () => {
```

Comment:
Maybe extend this test case to test "state 1, state 1, state 2" so it can verify whether the "state 1 -> state 2" event tracks duration correctly?

Reply:
good idea, have updated